pkg/diagnose/firewall.go

```diff
@@ -118,3 +135,164 @@ func getActiveGatewayNodeName(clusterInfo *cluster.Info, hostname string, status
 
 	return ""
 }
+
+func getGatewayIP(clusterInfo *cluster.Info, localClusterID string, status reporter.Interface) string {
+	gateways, err := clusterInfo.GetGateways()
+	if err != nil {
+		status.Failure("Error retrieving gateways from cluster %q: %v", clusterInfo.Name, err)
+		return ""
```

Comment:
```suggestion
		return "", status.Error(err, "Error retrieving gateways from cluster %q", clusterInfo.Name)
```

And similar in other places.